unixfs/benches/ingest-linux-tar.rs

```diff
@@ -0,0 +1,118 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+
+pub fn criterion_benchmark(c: &mut Criterion) {
+    let tar_bytes = std::fs::read("linux-5.6.tar").expect("read failed");
+    c.bench_function("ingest-linux-tar", |b| b.iter(|| ingest_tar(&tar_bytes)));
+}
+
+fn ingest_tar(bytes: &[u8]) {
+    use cid::Cid;
+    use ipfs_unixfs::dir::builder::{BufferingTreeBuilder, TreeOptions};
+    use ipfs_unixfs::file::adder::FileAdder;
+    use std::io::Read;
+
+    let mut buffer = Vec::new();
+
+    let mut archive = tar::Archive::new(std::io::Cursor::new(bytes));
+    let mut entries = archive.entries().unwrap();
+
+    let mut opts = TreeOptions::default();
+    opts.wrap_with_directory();
+    let mut tree = BufferingTreeBuilder::new(opts);
+
+    while let Some(entry) = entries.next() {
+        let mut entry = entry.unwrap();
+        let path = std::str::from_utf8(&*entry.path_bytes())
+            .unwrap()
+            .to_string(); // need to get rid of this
```

Comment:
You could probably just pre-allocate it in `criterion_benchmark` and pass it to `ingest_tar` as parameter.

Reply:
This seems to work.